addon/mixins/authenticated-route-mixin.js

```diff
@@ -4,7 +4,33 @@ import { assert } from '@ember/debug';
 import { computed } from '@ember/object';
 import { getOwner } from '@ember/application';
 import Configuration from './../configuration';
-import isFastBoot from 'ember-simple-auth/utils/is-fastboot';
+import isFastBootCPM, { isFastBoot } from '../utils/is-fastboot';
+
+/**
+ * If the user is unauthenticated, invoke `callback`
+ *
+ * @param {ApplicationInstance} owner The ApplicationInstance that owns the service (and possibly fastboot and cookie) service(s)
+ * @param {Transition} transition Transition for the user's original navigation
+ * @param {(...args: []any) => any} callback Callback that will be invoked if the user is unauthenticated
+ */
+function runIfUnauthenticated(owner, transition, callback) {
```

Comment:
Once we have more of the simple-auth without mixins (#1619) story in place, we can start to carefully name these functions and export them from their respective modules. For now, we may want to leave these as module-private, and make a single "designing the public API" pass at the end

Reply:
yep, I guess that makes sense. That will also probably be the hardest part about this - making sure ESA is still - despite to the relatively large number of things it does now - relatively easy to consume/integrate.